### DISCUSSION TITLE
Dimmer integrate as a slider

### DISCUSSION THREAD

**Discussion by nbalcaen** [joleys/niko-home-control-II#122] [Q&A] [answered]
Perhaps it's a stupid question, but would it be possible to integrate the dimmers with an additional slider from 0-100%?
Now they are added as a simple switch without the functionality to dim.

Or is there a way to use HA to change the dimming per dimmer? I have 6 dimmers type 551-72201 (easywave protocol)

**Comment by tijsverkoyen** [comment]
That is weird, I have also some dimmers and these are exposed as dimmers.

Could you provide me with a device list, and the name of the Switch in Home Assistant? See: https://github.com/joleys/niko-home-control-II?tab=readme-ov-file#i-need-to-see-the-device-list

**Reply by nbalcaen** [reply]
Received device list: {'Params': [{'Devices': [{'Uuid': 'f791f75e-0908-4493-809e-e376620b8e28', 'Identifier': '7872b13d-03eb-4786-bd96-6f282c479de4', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Inkom', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Inkom'}, {'LocationId': '07a8a620-28e9-497e-8bbf-4c35eccd81da'}, {'LocationIcon': 'prayer-room'}, {'IconCode': 'lamp2'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': '691ffed1-6a74-4eab-9c69-03e9592fd2c4', 'Identifier': '1SAG3100188270', 'Model': 'nhc', 'Online': 'False', 'Name': 'electricitymeter', 'Technology': 'p1', 'Type': 'electricitymeter', 'Traits': [], 'Parameters': [{'Segment': 'Central'}]}, {'Uuid': '981c296d-7f7f-4486-8b7a-153cdadb160f', 'Identifier': '190c8d70-1da1-4523-9f99-973732917bf2', 'Model': 'dimmer', 'Online': 'True', 'Name': 'Lamp Gang +2', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Gang +2'}, {'LocationId': 'a5c19a7a-de04-41c5-a79c-46b832746e4e'}, {'LocationIcon': 'stairways'}, {'IconCode': 'dimmable-lamp'}], 'Properties': [{'Aligned': 'True'}, {'Status': 'Off'}, {'Brightness': '70'}], 'PropertyDefinitions': [{'Aligned': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Boolean'}}, {'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}, {'Brightness': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Range(0.00,100.00,1.00)'}}]}, {'Uuid': 'e0e96081-5fb1-4d80-8f5e-2c4db242600f', 'Identifier': 'db554a27-6d56-419f-b2e4-b9b80bdc52ba', 'Model': 'dimmer', 'Online': 'True', 'Name': 'Lamp Eetplaats', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Eetplaats'}, {'LocationId': '79de3259-4c98-49e7-9a95-34cd58ad89fa'}, {'LocationIcon': 'dining-room'}, {'IconCode': 'dimmable-lamp'}], 'Properties': [{'Aligned': 'True'}, {'Status': 'On'}, {'Brightness': '100'}], 'PropertyDefinitions': [{'Aligned': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Boolean'}}, {'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}, {'Brightness': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Range(0.00,100.00,1.00)'}}]}, {'Uuid': '136ff93f-fc06-4121-82b4-9a0c5b17f58c', 'Identifier': '3d8539e0-cc1c-40af-b4a4-09f4ca5049f6', 'Model': 'light', 'Online': 'True', 'Name': 'Keukeneiland', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Keuken'}, {'LocationId': '802cd5d1-af3a-4cf7-b01d-0e03908dcf24'}, {'LocationIcon': 'kitchen'}, {'IconCode': 'lamp2'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': '3cc69900-b66a-4cd8-b42a-347915d3bad5', 'Identifier': '9a3d1218-1df7-4131-b7ed-018e2fc7f3e8', 'Model': 'dimmer', 'Online': 'True', 'Name': 'Muurspots', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Keuken'}, {'LocationId': '802cd5d1-af3a-4cf7-b01d-0e03908dcf24'}, {'LocationIcon': 'kitchen'}, {'IconCode': 'dimmable-lamp'}], 'Properties': [{'Aligned': 'True'}, {'Status': 'On'}, {'Brightness': '100'}], 'PropertyDefinitions': [{'Aligned': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Boolean'}}, {'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}, {'Brightness': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Range(0.00,100.00,1.00)'}}]}, {'Uuid': 'd0052a9b-3b11-460c-9414-8bb7640dfd7a', 'Identifier': 'e5e41a58-f151-412f-8e19-65b801a3a381', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Zolderkamer Tuin', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'IconCode': 'lamp2'}, {'LocationId': 'e3b4a9cb-bdd5-4d4b-9029-45ff4f5d319b'}, {'LocationIcon': 'bedroom-single'}, {'LocationName': 'Zolderkamer Tuin'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': '99a9c37f-21c9-4fda-90f4-356f8d12a32a', 'Identifier': 'ae7ca10a-6460-4b32-bd9b-98e2d461e316', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Badkamer 1', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Badkamer'}, {'LocationId': '60a201f1-961a-4e38-a054-d0890cc1b70d'}, {'LocationIcon': 'bathroom'}, {'IconCode': 'lamp2'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': '6aa98371-e2c3-46f6-ae33-a2d5d27fa7db', 'Identifier': '73b2482b-4906-4379-a12b-bcad058df682', 'Model': 'generic', 'Online': 'True', 'Name': 'energyhome', 'Technology': 'nikohomecontrol', 'Type': 'energyhome', 'Traits': [], 'Parameters': [{'PeakEstimations': '20'}, {'PeakPercentile': '90'}, {'PeakModeOptimizedPrediction': 'False'}, {'PeakMode': 'Direct'}, {'PeakAggregation': 'Maximum'}], 'Properties': [{'WaterVolume': ''}, {'ElectricalPowerFromGrid': '0.000'}, {'ElectricalPeakPowerToGrid': ''}, {'ElectricalEnergyFromGrid': ''}, {'ElectricalEnergyToGridCost': ''}, {'ElectricalEnergyToGrid': ''}, {'ElectricalEnergyProduction': ''}, {'ElectricalPowerToGrid': '0.000'}, {'ElectricalEnergyFromGridCost': ''}, {'ReportInstantUsage': 'False'}, {'GasCost': ''}, {'ElectricalEnergySelfConsumption': ''}, {'ElectricalEnergyConsumption': ''}, {'ElectricalMonthlyPeakPowerFromGrid': ''}, {'ElectricalPeakPowerFromGrid': ''}, {'GasVolume': ''}, {'WaterCost': ''}], 'PropertyDefinitions': [{'WaterVolume': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.0000,999999.9990,0.0001)'}}, {'ElectricalPowerFromGrid': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalPeakPowerToGrid': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalEnergyFromGrid': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalEnergyToGridCost': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.00000,999999999.00000,0.00001)'}}, {'ElectricalEnergyToGrid': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalEnergyProduction': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalPowerToGrid': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalEnergyFromGridCost': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.00000,999999999.00000,0.00001)'}}, {'ReportInstantUsage': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Boolean'}}, {'GasCost': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.00000,999999999.00000,0.00001)'}}, {'ElectricalEnergySelfConsumption': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalEnergyConsumption': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalMonthlyPeakPowerFromGrid': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'ElectricalPeakPowerFromGrid': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.000,999999999.000,0.001)'}}, {'GasVolume': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.0000,999999.9990,0.0001)'}}, {'WaterCost': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.00000,999999999.00000,0.00001)'}}]}, {'Uuid': '445c2d8f-c447-43fe-bbe9-432966e83ae4', 'Identifier': 'bd4c3a26-dcd9-41ab-adbf-4d485095af72', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Hoofdslaapkamer', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Hoofdslaapkamer'}, {'LocationId': '1ffeabe4-213b-4f9e-b1de-f4b7e91950b6'}, {'LocationIcon': 'bedroom'}, {'IconCode': 'lamp2'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': '9c5d21ea-4094-4593-9a34-2f75f5c8041d', 'Identifier': 'gatewayfw', 'Model': 'generic', 'Online': 'True', 'Name': 'gatewayfw', 'Technology': 'firmware', 'Type': 'gatewayfw', 'Traits': [{'HubType': 'SMARTBOX'}], 'Parameters': [], 'Properties': [{'ErrorCode': ''}, {'UpgradeStatus': 'Idle'}, {'CurrentFWInfo': '2.18.1.603 | 2023.1-20231006151344'}, {'AvailableFWInfo': ''}, {'RemoteAvailableFWInfo': ''}], 'PropertyDefinitions': [{'ErrorCode': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Text'}}, {'UpgradeStatus': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(Idle,RemoteVersionCheck,RemoteVersionVerify,DownloadImage,VerifyImage,WaitForConfirmation,UserConfirmed,Installing,InstallationCompleted,Reboot,WaitForUpload,UpgradingSubsystems,FallBack,FellBack)'}}, {'CurrentFWInfo': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Text'}}, {'AvailableFWInfo': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Text'}}, {'RemoteAvailableFWInfo': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Text'}}]}, {'Uuid': 'cbdcc524-95cb-4b54-97d7-5b0870c08e76', 'Identifier': '213ae5ae-1abf-4530-b085-5822538fb7e6', 'Model': 'dimmer', 'Online': 'True', 'Name': 'Lamp Gang +1', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Gang +1'}, {'LocationId': '75b25d95-b30a-404d-919e-fcd5025993d9'}, {'LocationIcon': 'stairways'}, {'IconCode': 'dimmable-lamp'}], 'Properties': [{'Aligned': 'True'}, {'Status': 'On'}, {'Brightness': '40'}], 'PropertyDefinitions': [{'Aligned': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Boolean'}}, {'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}, {'Brightness': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Range(0.00,100.00,1.00)'}}]}, {'Uuid': '232acd9d-2c66-45aa-9907-e7400bc5e623', 'Identifier': '20f1067e-653f-4c58-82f1-4afec4d6a7d4', 'Model': 'dimmer', 'Online': 'True', 'Name': 'Lamp Badkamer 2', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Badkamer'}, {'LocationId': '60a201f1-961a-4e38-a054-d0890cc1b70d'}, {'LocationIcon': 'bathroom'}, {'IconCode': 'dimmable-lamp'}], 'Properties': [{'Aligned': 'True'}, {'Status': 'Off'}, {'Brightness': '40'}], 'PropertyDefinitions': [{'Aligned': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Boolean'}}, {'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}, {'Brightness': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Range(0.00,100.00,1.00)'}}]}, {'Uuid': '699f25a5-778b-4e19-a9e7-96c4895b7b7c', 'Identifier': '0711c8d7-bb3b-456a-87f2-793a5a080dd3', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Bureau', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Bureau'}, {'LocationId': '68ad96de-b8b4-4286-9c1c-b365a612a678'}, {'LocationIcon': 'library'}, {'IconCode': 'lamp2'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': 'be4adff1-afa3-4747-9aeb-25c7dcb69732', 'Identifier': '5e01bd3b-f7c2-4586-aba2-0af1da1d46c3', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Gang +0', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Gang +0'}, {'LocationId': 'f1454122-e6f1-4cf1-b24a-1bd162501441'}, {'LocationIcon': 'lobby'}, {'IconCode': 'lamp2'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': '9c2187d0-29e4-43fb-bc16-f103e8533009', 'Identifier': '645bbf1d-9965-4ceb-abea-2cf7ad4260ef', 'Model': 'light', 'Online': 'True', 'Name': 'Lamp Thuis', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'IconCode': 'lamp2'}, {'LocationId': 'a19c72ad-ff31-4c0f-976c-c3259b664bc8'}, {'LocationIcon': 'bedroom-single'}, {'LocationName': 'Zolderkamer Straat'}], 'Properties': [{'Status': 'Off'}], 'PropertyDefinitions': [{'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}]}, {'Uuid': 'd7488e6f-13ef-4e23-af25-c270adb20228', 'Identifier': '10a7b419-e2a9-4080-954c-138828fc7d75', 'Model': 'dimmer', 'Online': 'True', 'Name': 'Lamp Zithoek', 'Technology': 'nikohomecontrol', 'Type': 'action', 'Traits': [], 'Parameters': [{'LocationName': 'Zithoek'}, {'LocationId': '40e95e0d-0f2e-420e-9691-0bdcfd571ad9'}, {'LocationIcon': 'living'}, {'IconCode': 'dimmable-lamp'}], 'Properties': [{'Aligned': 'True'}, {'Status': 'On'}, {'Brightness': '82'}], 'PropertyDefinitions': [{'Aligned': {'HasStatus': 'true', 'CanControl': 'false', 'Description': 'Boolean'}}, {'Status': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Choice(On,Off)'}}, {'Brightness': {'HasStatus': 'true', 'CanControl': 'true', 'Description': 'Range(0.00,100.00,1.00)'}}]}]}], 'Method': 'devices.list'}

**Comment by nbalcaen** [comment]
Uit de device list leid ik af dat de integratie de dimmers wel herkend, maar bij de bediening zijn deze weergegeven als 'eenvoudige' schakelaar. Dien ik iets aan te passen?

![image](https://github.com/joleys/niko-home-control-II/assets/127793636/bbfcddc6-81ff-44cd-b852-7f59e253091c)

**Comment by nbalcaen** [comment, marked as answer]
OK, problem solved.

**Comment by tijsverkoyen** [comment]
@nbalcaen what was the problem? And how did you resolve it?

**Reply by nbalcaen** [reply]
It was actually functioning from the beginning.
